src/pb_stub.cc

```diff
@@ -402,7 +470,40 @@ Stub::Initialize(bi::managed_external_buffer::handle_t map_handle)
           .attr("TritonPythonModel");
   deserialize_bytes_ = python_backend_utils.attr("deserialize_bytes_tensor");
   serialize_bytes_ = python_backend_utils.attr("serialize_byte_tensor");
-  model_instance_ = TritonPythonModel();
+
+  return TritonPythonModel();
+}
+
+void
+Stub::AutoCompleteModelConfig(
+    bi::managed_external_buffer::handle_t string_handle,
+    std::string* auto_complete_config)
+{
+  model_ = StubSetup();
+
+  std::unique_ptr<PbString> pb_string_shm =
+      PbString::LoadFromSharedMemory(shm_pool_, string_handle);
+
+  py::module python_backend_utils =
+      py::module_::import("triton_python_backend_utils");
+  py::object model_config =
+      python_backend_utils.attr("ModelConfig")(pb_string_shm->String());
+
+  if (py::hasattr(model_, "auto_complete_config")) {
+    model_config = model_.attr("auto_complete_config")(model_config);
+  }
+  if (py::isinstance<py::none>(model_config)) {
+    throw PythonBackendException(
+        "auto_complete_config function in model '" + name_ +
+        "' must return a valid pb.ModelConfig object.");
+  }
+  (*auto_complete_config) = std::string(py::str(model_config));
```

Comment:
I think the reason that you need the FixStringToJSON in python_be.cc is that you are not really converting the dictionary to a json here. You should be calling json.dumps to convert the dictionary to a JSON: https://docs.python.org/3/library/json.html#json.dumps

Reply:
Call json.dumps here.

Reply:
Please see my comment: https://github.com/triton-inference-server/python_backend/pull/138#discussion_r881901906